ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousPlannerSchedulingTask.java

```diff
@@ -135,6 +140,18 @@ public ContinuousPlannerSchedulingTask(DRCRobotModel robotModel,
 
       // Added a couple listeners to help when jumping between states
       stateMachine = stateMachineFactory.build(ContinuousHikingState.DO_NOTHING);
+
+      stateMachine.addPreTransitionCallback(() ->
+                                            {
+                                               if (controllerFootstepQueueMonitor.getRobotFalling() || resetStateMachine.get())
+                                               {
+                                                  LogTools.info("---- Resetting State Machine for Continuous Hiking ----");
+                                                  stateMachine.resetToInitialState();
+                                                  resetStateMachine.set(false);
+                                                  commandMessage.get().setEnableContinuousHiking(false);
+                                               }
```

Comment:
More atomic usage of AtomicBoolean
```suggestion
                                               if (controllerFootstepQueueMonitor.getRobotFalling() || resetStateMachine.getAndSet(false))
                                               {
                                                  LogTools.info("---- Resetting State Machine for Continuous Hiking ----");
                                                  stateMachine.resetToInitialState();                                      
                                                  commandMessage.get().setEnableContinuousHiking(false);
                                               }
```

Reply:
I left this as is